packages/devtools_app/lib/src/url_utils.dart

```diff
@@ -27,11 +27,21 @@ String getSimplePackageUrl(String url) {
   return url;
 }
 
-/// Returns a trimmed vm service uri without any trailing characters.
+/// Returns a normalized vm service uri.
+///
+/// Removes trailing characters, trailing url fragments, and decodes urls that
+/// were accidentally encoded.
 ///
 /// For example, given a [value] of http://127.0.0.1:60667/72K34Xmq0X0=/#/vm,
 /// this method will return the URI http://127.0.0.1:60667/72K34Xmq0X0=/.
-Uri getNormalizedTrimmedUri(String value) {
+Uri normalizeVmServiceUri(String value) {
+  value = value.trim();
+
+  // Cleanup encoded urls likely copied from the uri of an existing running
+  // DevTools app.
+  if (value.contains('%3A%2F%2F')) {
```

Comment:
What's the deal with this?

Reply:
// Cleanup encoded urls likely copied from the uri of an existing running
 // DevTools app.

This is when someone copies the url from an existing devtools URL e.g.
http://localhost:8080?uri=%3A%2F%2Flocalhost:8080/asdfaer

This check was in the preview devtools app but got lost because it was in the dart:html specific code instead of the shared business logic.

Reply:
I see. What's the problem with decoding an already decoded value? If it's a no-op then we can simply remove the conditional.